src/ecosystem/Ecosystem.jl

```diff
@@ -44,12 +44,15 @@ function map_to_discrete(v::Union{Int64,Float64}, u::Union{Int64,Float64})::Int6
 end
 
 """
-    map_to_discrete!(df::DataFrame, u::AbstractArray)::Nothing
+    map_to_discrete!(df::Union{DataFrame,SubDataFrame}, u::Union{AbstractVector{Union{Int64,Float64}},Tuple})::Nothing
```

Comment:
I haven't come across a SubDataFrame type before- does this include things like DataFrameRow?